src/persist/src/indexed/mod.rs

```diff
@@ -1586,4 +1605,43 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_dropped_listener() -> Result<(), Error> {
+        let mut i = MemRegistry::new().indexed_no_reentrance()?;
+        let id = block_on(|res| i.register("0", "()", "()", res))?;
+
+        let mut listen_rxs = vec![];
+
+        for _ in 0..5 {
+            // Register a listener for writes.
+            let (listen_tx, listen_rx) = crossbeam_channel::unbounded();
+            block_on(|res| i.listen(id, listen_tx, res))?;
+            listen_rxs.push(listen_rx);
+        }
+
+        let mut expected = vec![];
+        for t in 0..5 {
+            // Write data at even times and seal at odd times, to exercise dropping listeners
+            // while notifying them of writes and notifying them of seals.
+            if t % 2 == 0 {
+                let updates: Vec<((Vec<u8>, Vec<u8>), u64, isize)> =
+                    vec![((format!("{}", t).into(), "".into()), t, 1)];
+                block_on_drain(&mut i, |i, res| {
+                    i.write(vec![(id, updates.iter().collect::<ColumnarRecords>())], res)
+                })?;
+                expected.push(ListenEvent::Records(updates));
+            } else {
+                block_on_drain(&mut i, |i, res| i.seal(vec![id], t, res))?;
+                expected.push(ListenEvent::Sealed(t));
+            }
+
+            let listen_rx = listen_rxs.pop().expect("known to have 5 listeners");
```

Comment:
instead of a pop().expect here, let's make this whole bit a for loop over listen_rxs.into_iter().enumerate()